packages/batch-tx-handler/src/l1-batch-handler/utils.ts

```diff
@@ -107,21 +147,21 @@ export const decodeL2Msgs = (l2Msgs: Uint8Array): string[] => {
       txHash.push(...decodeL2Msgs(nextData));
       current = endOfNext;
     }
+  } else if (kind === delayedMsgToBeAdded) {
+    const remainData: Uint8Array = l2Msgs.subarray(1);
+    const currentHash = ethers.utils.hexlify(remainData);
+    txHash.push(currentHash);
   }
   return txHash;
 };
 
 // Get related sequencer batch data from a sequencer batch submission transaction.
-export const getRawData = async (
-  sequencerTx: string,
-  l2NetworkId: number,
-  provider: ethers.providers.JsonRpcProvider,
-): Promise<Uint8Array> => {
+export const getRawData = async (sequencerTx: string): Promise<[Uint8Array, BigNumber]> => {
   //Because current arbitrum-sdk doesn't support latest sequencer inbox contract, so we use ethersjs here directly.
   const contractInterface = new Interface(seqFunctionAbi);
   const l2Network = await getL2Network(l2NetworkId);
-  const txReceipt = await provider.getTransactionReceipt(sequencerTx);
-  const tx = await provider.getTransaction(sequencerTx);
+  const txReceipt = await l1Provider.getTransactionReceipt(sequencerTx);
+  const tx = await l1Provider.getTransaction(sequencerTx);
   if (!tx || !txReceipt || (txReceipt && !txReceipt.status)) {
```

Comment:
```suggestion
  if (!tx || !txReceipt || !txReceipt.status) {
```
This evaluates from left to right, so you don't need to check for txReceipt again (because you already did in the second condition).